src/test/java/de/tum/in/www1/artemis/StudentQuestionIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+package de.tum.in.www1.artemis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.ZonedDateTime;
+import java.util.HashSet;
+import java.util.List;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.domain.Attachment;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.Lecture;
+import de.tum.in.www1.artemis.domain.StudentQuestion;
+import de.tum.in.www1.artemis.repository.*;
+import de.tum.in.www1.artemis.service.StudentQuestionService;
+import de.tum.in.www1.artemis.util.DatabaseUtilService;
+import de.tum.in.www1.artemis.util.ModelFactory;
+import de.tum.in.www1.artemis.util.RequestUtilService;
+
+public class StudentQuestionIntegrationTest extends AbstractSpringIntegrationTest {
+
+    @Autowired
+    DatabaseUtilService database;
+
+    @Autowired
+    RequestUtilService request;
+
+    @Autowired
+    CourseRepository courseRepo;
+
+    @Autowired
+    ExerciseRepository exerciseRepo;
+
+    @Autowired
+    LectureRepository lectureRepo;
+
+    @Autowired
+    AttachmentRepository attachmentRepo;
+
+    @Autowired
+    StudentQuestionRepository studentQuestionRepository;
+
+    @Autowired
+    StudentQuestionService studentQuestionService;
+
+    @BeforeEach
+    public void initTestCase() {
+        database.addUsers(5, 5, 1);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        database.resetDatabase();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void createStudentQuestion() throws Exception {
+        StudentQuestion studentQuestion = new StudentQuestion();
+        studentQuestion.setQuestionText("Test Student Question 1");
+        studentQuestion.setVisibleForStudents(true);
+
+        StudentQuestion createdStudentQuestion = request.postWithResponseBody("/api/student-questions", studentQuestion, StudentQuestion.class, HttpStatus.CREATED);
+
+        assertThat(createdStudentQuestion).isNotNull();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void createExistingStudentQuestion() throws Exception {
+        StudentQuestion studentQuestion = database.createCourseWithExerciseAndStudentQuestions().get(0);
+
+        request.postWithResponseBody("/api/student-questions", studentQuestion, StudentQuestion.class, HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void editStudentQuestion() throws Exception {
```

Comment:
Can another student edit this student's question?

Reply:
Same as with the answers, everyone can edit it, but I still added an additional case for it.

Reply:
Students should **only** be able to edit and delete their **own** questions. They should **NOT** be able to edit and delete questions of other users.

Tutors and instructors should be able to edit and delete questions of other users.

Please implement this logic in the tests and let them fail if the server logic is not implemented correctly, then we should fix it!